doc/en/user/source/production/config.rst

```diff
@@ -49,39 +49,44 @@ Make sure clients cannot request an inordinate amount of resources from your ser
 
 In particular:
 
-* Set the maximum amount of features returned by each WFS GetFeature request (this can also be set on a per featuretype basis by modifying the :file:`info.xml` files directly)
-* Set the WMS ``request limits`` so that no request will consume too much memory or too much time
+* Set the :ref:`maximum amount of features <services_webadmin_wfs>` returned by each WFS GetFeature request (this can also be set on a per featuretype basis by modifying the :ref:`layer publishing wfs settings <data_webadmin_layers>`).
+* Set the :ref:`WMS request limits <wms_configuration>` so that no request will consume too much memory or too much time
+* Set :ref:`WPS limits <webadmin_wps>`, so no process will consume too much memory or too much time. You may also limit the :ref:`size input parameters <wps_security>` for further control.
 
-Set security
-------------
+Set security for data modification
+----------------------------------
 
-GeoServer includes support for WFS-T (transactions) by default, which lets users modify your data. If you don't want your database modified, you can turn off transactions in the the :ref:`web_admin`. Set the :guilabel:`Service Level` to ``Basic``.
+GeoServer includes support for WFS-T (transactions) by default, which lets users modify your data.
 
-If you'd like some users to be able to modify some but not all of your data, you will have to set up an external security service. An easy way to accomplish this is to run two GeoServer instances and configure them differently, and use authentication to only allow certain users to have access.
+If you don't want your database modified, you can turn off transactions in the the :ref:`services_webadmin_wfs`. Set the :guilabel:`Service Level` to ``Basic``. For extra security, we recommend any database access use datastore credentials providing read-only permissions. This will eliminate the possibility of a SQL injection (though GeoServer is generally not vulnerable to that sort of attack).
 
-For extra security, make sure that the connection to the datastore that is open to all is through a user who has read-only permissions. This will eliminate the possibility of a SQL injection (though GeoServer is generally not vulnerable to that sort of attack).
+If you would like some users to be able to modify data, set the service level :guilabel:`Service Level` to ``Transactional`` (or ``Complete``) and use :ref:`security_service` to limit access to the `WFS.Transaction` operation.
+
+If you would like some users to be able to modify some but not all of your data, set the :guilabel:`Service Level` to ``Transactional`` (or ``Complete``), and use :ref:`security_layer` to limit write access to specific layers. Data security can be used to allow write access based on a to a workspace, datastore, or layer security.
```

Comment:
If you would like some users to be able to modify some but not all of your data, set the :guilabel:`Service Level` to ``Transactional`` (or ``Complete``), and use :ref:`security_layer` to limit write access to specific layers. Data security can be used to allow write access based on workspace, datastore, or layer security.